bin/alluxio-mount.sh

```diff
@@ -87,6 +87,65 @@ function umount_ramfs_linux() {
   fi
 }
 
+function check_space_freebsd() {
+  local total_mem=$(($(sysctl -h hw.usermem | awk 'NR==1{print $2}' | sed 's/,//g')))
+  if [[ ${total_mem} -lt ${MEM_SIZE} ]]; then
+    echo "ERROR: Memory(${total_mem}) is less than requested ramdisk size(${MEM_SIZE}). Please
+    reduce alluxio.worker.memory.size in alluxio-site.properties" >&2
+    exit 1
+  fi
+}
+
+function mount_ramfs_freebsd() {
+  TIER_PATH=${1}
+  echo "Formatting RamFS: ${TIER_PATH} (${MEM_SIZE})"
+  if [[ ${USE_SUDO} == true ]]; then
+    sudo mkdir -p ${TIER_PATH}
+  else
+    mkdir -p ${TIER_PATH}
+  fi
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR: mkdir ${TIER_PATH} failed" >&2
+    exit 1
+  fi
+
+  if [[ ${USE_SUDO} == true ]]; then
+    sudo mount -t tmpfs -o size=${MEM_SIZE} tmpfs ${TIER_PATH}
```

Comment:
is `tmpfs` vs `ramfs` the only difference between this and the linux implementation?

Reply:
Additionally, I came across this link in the freebsd handbook which suggests a different approach: https://www.freebsd.org/doc/handbook/disks-virtual.html

> To instead create a new memory-based memory disk with mdmfs, use this one command:
```
# mdmfs -s 5m md1 /mnt
```

Here is the man page on `mdfs`: www.freebsd.org/cgi/man.cgi?query=mdmfs&sektion=8&manpath=freebsd-release-ports

Have you considered this option?

Reply:
> Additionally, I came across this link in the freebsd handbook which suggests a different approach: https://www.freebsd.org/doc/handbook/disks-virtual.html
> 
> > To instead create a new memory-based memory disk with mdmfs, use this one command:
> 
> ```
> # mdmfs -s 5m md1 /mnt
> ```
> 
> Here is the man page on `mdfs`: [www.freebsd.org/cgi/man.cgi?query=mdmfs&sektion=8&manpath=freebsd-release-ports](http://www.freebsd.org/cgi/man.cgi?query=mdmfs&sektion=8&manpath=freebsd-release-ports)
> 
> Have you considered this option?

Yes I did but I wanted the code to be as similar to the linux version as possible. The mount -t tmpfs does basically the same thing based on the alluxio usage, see https://www.freebsd.org/cgi/man.cgi?query=mdmfs&sektion=8&manpath=freebsd-release-ports:

```
     When md-device is `auto', mdmfs uses tmpfs(5) if it is present in the
     kernel or can be loaded as	a module, otherwise it falls back to using
     md(4) auto-unit as	if `md'	had been specified.

     When md-device is `tmpfs',	mdmfs mounts a tmpfs(5)	filesystem, translat-
     ing the -s	size option, if	present, into a	`-o size=' mount option.  Any
     -o	options	on the command line are	passed through to the tmpfs(5) mount.
     Options specific to mdconfig(8) or	newfs(8) are ignored.
```

> is `tmpfs` vs `ramfs` the only difference between this and the linux implementation?

The other difference is that this implementation also uses sysctl instead of /proc/meminfo to avoid requiring mounting procfs.

Reply:
Thanks for the clarification!